### PR TITLE
Moodle::Login.moodle_login: fix login success verification regex

### LINKED ISSUE #19226: Msf::Exploit::Remote::HTTP::Moodle::Login.moodle_login: misses valid logins
The `Msf::Exploit::Remote::HTTP::Moodle::Login.moodle_login` function relies upon page `title` HTML tags to determine whether the login was successful.

https://github.com/rapid7/metasploit-framework/blob/a92e37ecfb916ed5a786b9920ec77ff0c42ed5d0/lib/msf/core/exploit/remote/http/moodle/login.rb#L26-L28

The following page titles are returned by Bitnami and TurnkeyLinux moodle deployments respectively:

```html
<title>Dashboard | New Site</title>
```

```html
<title>Dashboard | moodle</title>
```

This HTML does not match the regex.

For example:

```diff
diff --git a/lib/msf/core/exploit/remote/http/moodle/login.rb b/lib/msf/core/exploit/remote/http/moodle/login.rb
index cf44add997..960e8eddc1 100644
--- a/lib/msf/core/exploit/remote/http/moodle/login.rb
+++ b/lib/msf/core/exploit/remote/http/moodle/login.rb
@@ -23,6 +23,8 @@ module Msf::Exploit::Remote::HTTP::Moodle::Login
       'keep_cookies' => true
     }, timeout, 20) # typical redirect is 3-5, but it may do more if caching gets messed up on server
 
+    puts res.body.scan(/<title>(.*?)<\/title>/).flatten.first
+
     if !res || (res.code != 200) || !res.body.include?('<title>Dashboard</title>')
       return nil
     end
```

```
msf6 auxiliary(scanner/http/moodle_login) > run

[+] 192.168.200.141:443 - Found Moodle
Dashboard | moodle   # <-- valid password page title
[-] 192.168.200.141:443 - [1/1] - Failed: 'admin:Password1'
[!] No active DB -- Credential data will not be saved!
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/moodle_login) > set username notadmin
username => notadmin
msf6 auxiliary(scanner/http/moodle_login) > run

[+] 192.168.200.141:443 - Found Moodle
Log in to the site | moodle   # <-- invalid password page title
[-] 192.168.200.141:443 - [1/1] - Failed: 'notadmin:Password1'
[!] No active DB -- Credential data will not be saved!
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/moodle_login) > 
```

You can download moodle virtual machines here:

* https://bitnami.com/stack/moodle/virtual-machine
* https://www.turnkeylinux.org/moodle
* https://docs.moodle.org/404/en/Moodle_appliance

### PR DESCRIPTION
**Pull request by bcoles**
Fixes #19226

### CODE REVIEW

File: lib/msf/core/exploit/remote/http/moodle/login.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def moodle_login(user, pass, timeout = 20)
       'keep_cookies' => true
     }, timeout, 20) # typical redirect is 3-5, but it may do more if caching gets messed up on server
 
-    if !res || (res.code != 200) || !res.body.include?('<title>Dashboard</title>')
+    if !res || (res.code != 200) || !res.body.include?('<title>Dashboard')
```
Comment (jvoisin) [inline lib/msf/core/exploit/remote/http/moodle/login.rb:26]:
```suggestion
    # eg. `<title>Dashboard | moodle</title>` or `<title>Dashboard | New Site</title>`
    if !res || (res.code != 200) || !res.body.include?('<title>Dashboard [|] ')
```

Comment (bcoles) [reply lib/msf/core/exploit/remote/http/moodle/login.rb:26]: This change does not match the existing regex.

**Comment by sjanusz-r7** [comment]
Seems legit! Thank you. 🚀 

After setting up the `bitnami` VM and playing around with the `moodle_login` function, the response I get is:
```ruby
res
=> #<Rex::Proto::Http::Response:0x000000010e140d98
 @auto_cl=false,
 @body=
  "<!DOCTYPE html>\n\n<html  dir=\"ltr\" lang=\"en\" xml:lang=\"en\">\n<head>\n    <title>Dashboard | New Site</title>\n
```
Inside of a `pry` session when receiving the response:

## Before
```ruby
res.body.include?('<title>Dashboard</title>')
=> false
```

## After
```ruby
res.body.include?('<title>Dashboard')
=> true
```

**Comment by sjanusz-r7** [comment]
## Release Notes
Fixed an issue in Moodle::Login.moodle_login that reported a false negative when logging in with user's credentials.

**Pull request merged**